### PR TITLE
ledger-on-sql: Run the indexer on PostgreSQL when possible.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
Specifically, when running the PostgreSQL conformance tests.

It will also work when running it as a binary, but you need to specify both the `--server-jdbc-url` and `--jdbc-url` switches. This is to keep _kvutils/app_ simple.

To do this, I needed to add a prefix of "ledger_" to the Flyway migrations table, to avoid conflicts with the index's migrations table. While I was at it, I made this a constant in the code and used Flyway placeholders to substitute it into the migration files. We don't have any users right now, so changing the migrations should be fine.

When we roll out the new schema for the index, I suggest we add a prefix of "index_" to all tables there, too.

This gets us a little closer to #4084.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/ExtraConfig.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.on.sql
+
+case class ExtraConfig(jdbcUrl: Option[String])
```
Comment (stefanobaghino-da) [inline ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/ExtraConfig.scala]:
```suggestion
final case class ExtraConfig(jdbcUrl: Option[String])
```

File: libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwner.scala [hunk 2]
```diff
@@ -48,13 +48,13 @@ trait ResourceOwner[+A] {
 
 object ResourceOwner {
   def successful[T](value: T): ResourceOwner[T] =
-    new FutureResourceOwner[T](() => Future.successful(value))
+    new FutureResourceOwner(() => Future.successful(value))
 
-  def failed(throwable: Throwable): ResourceOwner[Nothing] =
-    new FutureResourceOwner[Nothing](() => Future.failed(throwable))
+  def failed[T](throwable: Throwable): ResourceOwner[T] =
```
Comment (stefanobaghino-da) [inline libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwner.scala]: Why do you need this? `ResourceOwner` should be covariant now.

Comment (stefanobaghino-da) [reply libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwner.scala]: I see. I would suggest you leave this as is and either:
- hint the compiler of the type you want to fold to (`.fold[ResourceOwner[Config[Extra]]]`), or
- use `map` followed by `getOrElse`

In the first case you would end up with
```scala
  def owner[Extra](
      name: String,
      extraOptions: OptionParser[Config[Extra]] => Unit,
      defaultExtra: Extra,
      args: Seq[String],
  ): ResourceOwner[Config[Extra]] =
    parse(name, extraOptions, defaultExtra, args)
      .fold[ResourceOwner[Config[Extra]]](ResourceOwner.failed(new Config.ConfigParseException))(
        ResourceOwner.successful)
```
and in the second with
```scala
  def owner[Extra](
      name: String,
      extraOptions: OptionParser[Config[Extra]] => Unit,
      defaultExtra: Extra,
      args: Seq[String],
  ): ResourceOwner[Config[Extra]] =
    parse(name, extraOptions, defaultExtra, args)
      .map(ResourceOwner.successful)
      .getOrElse(ResourceOwner.failed(new Config.ConfigParseException))
```

Comment (SamirTalwar-DA) [reply libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwner.scala]: Yeah, I was not too happy with this but I think you're right. I'll revert that change.

Comment (SamirTalwar-DA) [reply libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwner.scala]: Done.

**Pull request merged**